test/integration/backward_compatible/parallel/map/NearCachedMapTest.js

```diff
@@ -102,9 +107,19 @@ describe('NearCachedMapTest', function () {
                 }
                 await map1.get('key1');
                 await map2.remove('key1');
-                const val = await TestUtil.promiseLater(5000, map1.get.bind(map1, 'key1'));
-                expectStats(map1, 0, 2, 1);
-                expect(val).to.be.null;
+                const mapProxyGetSpy = sandbox.replace(
+                    MapProxy.prototype, 'getInternal', sandbox.fake(MapProxy.prototype.getInternal)
+                );
+
+                let previousCallCount = mapProxyGetSpy.callCount;
+
+                await assertTrueEventually(async () => {
+                    previousCallCount = mapProxyGetSpy.callCount;
+                    const val = await map1.get('key1');
+                    // Client should eventually use near cache
+                    expect(mapProxyGetSpy.callCount).to.be.eq(previousCallCount);
+                    expect(val).to.be.null;
+                }, 1000);
```

Comment:
I think spying internals in a backward-compatible test should be avoided. What about a version like below?
```js
                await map1.get('key1');
                await map2.remove('key1');

                await assertTrueEventually(async () => {
                    const val = await map1.get('key1');
                    getNearCacheMissCount(map1).to.be.greaterThan(0); // Or whatever the syntax is
                    expect(val).to.be.null;
                }, 1000);
```

Reply:
```
                console.log(getNearCacheStats(map1).missCount);
                await map1.get('key1');
                console.log(getNearCacheStats(map1).missCount);
                await map2.remove('key1');

                console.log(getNearCacheStats(map1).missCount);

                await assertTrueEventually(async () => {
                    const val = await map1.get('key1');
                    // Client should eventually use near cache
                    console.log(getNearCacheStats(map1).missCount);
                    expect(val).not.to.be.null;
                }, 1000);
```

when I do this it logs:

```
0
1
1
and then
2
2
2
2
2
```

So first get() increases miss count by 1 so it is always more than 0. Also you were saying we should make sure that the client does not send invocation to cluster anymore and use near cache. So we should assert that hitCount increases. Am I right?

Reply:
New version of the test is in expect

Reply:
Then, I think it would be ok to make it `expect(getNearCacheStats(map1).hitCount).to.be.greaterThan(1);`. I just want to see that, we can't find that key in the near cache, because it is invalidated after some time.

Your version is also OK